### PR TITLE
bump vite to `6.0.0-beta.8`

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
> [!Warning]
> This PR is blocked by this issue: https://github.com/vitejs/vite/pull/18395/files#r1832458011
> Fixed by: https://github.com/vitejs/vite/pull/18611 (this will apply only in the next Vite beta 10 release though)

### CODE REVIEW

File: packages/vite-plugin-cloudflare/src/cloudflare-environment.ts
Comment (dario-piotrowicz) [inline packages/vite-plugin-cloudflare/src/cloudflare-environment.ts:143]: Vite's moved this option to the top level: https://github.com/vitejs/vite/pull/18465

File: packages/vite-plugin-cloudflare/src/cloudflare-environment.ts
Comment (dario-piotrowicz) [inline packages/vite-plugin-cloudflare/src/cloudflare-environment.ts:181]: They've also removed `webCompatible` and kind of replaced it with `keepProcessEnv` (https://github.com/vitejs/vite/pull/18395#discussion_r1820503185)

File: packages/vite-plugin-cloudflare/src/cloudflare-environment.ts
Comment (dario-piotrowicz) [inline packages/vite-plugin-cloudflare/src/cloudflare-environment.ts:181]: as far as I understand `keepProcessEnv: true` basically asks Vite not to mess with `process.env` which I think is something that we want ([source](https://github.com/vitejs/vite/blob/ce0eec91c9b5a91c52bf5b9bae6173d35f6e1649/packages/vite/src/node/config.ts#L246-L250))

**Comment by jamesopstad** [comment]
I think it's best to hold off upgrading until #49 is merged.

File: packages/vite-plugin-cloudflare/src/index.ts
Comment (dario-piotrowicz) [inline packages/vite-plugin-cloudflare/src/index.ts:37]: @petebacondarwin I needed to move this to the environment config since `conditions` is no longer inherited by environments ([source change](https://github.com/vitejs/vite/commit/d002e7d05a0f23110f9185b39222819bcdfffc16#diff-11e17761d4ecfee8f8fde15c6d79b7bc0260176396a30dfd8e6f6bbaf5af4745R682)) I hope this is fine, was there a particular reason why you added the `conditions` here?

Comment (petebacondarwin) [reply packages/vite-plugin-cloudflare/src/index.ts:37]: This was the place I found that worked. But if it works in the environment config that is also cool. 
Does that cover builds too?

**Comment by petebacondarwin** [comment]
Actually the CI jobs are not happy at all...

**Comment by dario-piotrowicz** [comment]
> Actually the CI jobs are not happy at all...

yes, things only pass with my local version of vite (unless I do apply a pnpm patch)

File: packages/vite-plugin-cloudflare/src/index.ts
Comment (dario-piotrowicz) [reply packages/vite-plugin-cloudflare/src/index.ts:37]: > Does that cover builds too?

It is a top-level configuration so I would definitely say yes 🤔

File: packages/vite-plugin-cloudflare/src/cloudflare-environment.ts
Comment (dario-piotrowicz) [inline packages/vite-plugin-cloudflare/src/cloudflare-environment.ts:169]: @petebacondarwin @jamesopstad I put `neutral` as that seemed the preference in chat? do you agree? or do you think that `browser` would be better?

Comment (jamesopstad) [reply packages/vite-plugin-cloudflare/src/cloudflare-environment.ts:169]: I think we should go with `neutral` as it seems more appropriate and then we can always change to `browser` later if needed.

File: packages/vite-plugin-cloudflare/src/cloudflare-environment.ts
Comment (jamesopstad) [reply packages/vite-plugin-cloudflare/src/cloudflare-environment.ts:184]: Interesting. Perhaps using `false` so that it replaces it with an empty object would be good when `nodejs_compat` is not enabled?

**Review by jamesopstad — APPROVED** [review]
Looks good to me!

File: packages/vite-plugin-cloudflare/src/cloudflare-environment.ts
Comment (dario-piotrowicz) [reply packages/vite-plugin-cloudflare/src/cloudflare-environment.ts:169]: sounds good to me 🙂

File: packages/vite-plugin-cloudflare/src/cloudflare-environment.ts
Comment (dario-piotrowicz) [reply packages/vite-plugin-cloudflare/src/cloudflare-environment.ts:184]: I don't have a super strong opinion on this one

I think I am a bit more inclined to say that `keepProcessEnv: true` is always the best in our case so that we completely leave the handling/initialization of `process.env` up to workerd

but again I am fine with either 🙂  (it should also be simple to revert this back later, although having to check if nodejs_compat is set is sound to force us to move this code around unfortunately I think (as I did here: https://github.com/flarelabs-net/vite-plugin-cloudflare/pull/52/files#diff-0217c39a7be6fcdb410ac8dfea2ae2998b882a1aaba3f2e83ff2831a0beee7bfR71-R73))

File: packages/vite-plugin-cloudflare/src/cloudflare-environment.ts
Comment (petebacondarwin) [reply packages/vite-plugin-cloudflare/src/cloudflare-environment.ts:169]: Also neutral adds in the fewest extra behaviours, which we can (mostly) add in manually if needed. The other modes require us to remove behaviours, which is generally more tricky.

File: packages/vite-plugin-cloudflare/src/cloudflare-environment.ts
Comment (petebacondarwin) [reply packages/vite-plugin-cloudflare/src/cloudflare-environment.ts:184]: I agree that keeping it set to true for now seems the best approach.

**Review by petebacondarwin — APPROVED** [review]

File: packages/vite-plugin-cloudflare/src/cloudflare-environment.ts
Comment (dario-piotrowicz) [reply packages/vite-plugin-cloudflare/src/cloudflare-environment.ts:184]: ah ok, let's go with that for now then (and we can always revisit later on if needed) 🙂

**Pull request merged**
